src/de/willuhn/jameica/hbci/server/UmsatzGroup.java

```diff
@@ -98,10 +98,8 @@ public GenericIterator getPossibleParents() throws RemoteException
    */
   public boolean hasChild(GenericObjectNode node) throws RemoteException
   {
-    for (int i=0;i<this.umsaetze.size();++i)
-    {
-      GenericObject o = (GenericObject) this.umsaetze.get(i);
-      if (o.equals(node))
+    for (Umsatz u : this.umsaetze) {
+      if (u.equals(node))
```

Comment:
Funktioniert der `equals` hier, oder muss `node` noch auf `Umsatz` gecastet werden?

Reply:
Ja, in [UmsatzImpl::equals](https://github.com/willuhn/hibiscus/blob/master/src/de/willuhn/jameica/hbci/server/UmsatzImpl.java#L300-L328) wird stets versucht, auf `Umsatz` zu casten:
```java
public boolean equals(GenericObject o) throws RemoteException {
		if (o == null || !(o instanceof Umsatz))
			return false;
		try
		{
			Umsatz other = (Umsatz) o;
...
```